app/src/app/sections/lia/lia-submission-page/lia-submission-page.component.ts

```diff
@@ -42,14 +51,78 @@ export class LiaSubmissionPageComponent implements OnInit, CanDeactivate {
 
     this.liaService.getUserLia(userId, liaId).then(res => {
       this.lia = res;
+      let remainingSeconds = this.getRemainingSeconds(this.lia);
 
       if (this.lia.state === 'submitted' || this.lia.state === 'opened') {
         let link = ['LiaLandingPage', { userId: userId, liaId: res.id }];
         this.router.navigate(link);
       }
+
+      // Start interval that saves progress.
+      if (this.lia.state === 'in_progress') {
+        this.startSaveProgressInterval();
+      }
+
+      // Check if time is up.
+      if (remainingSeconds <= 0) {
+        this.doAutoSubmit(this.lia);
+        return;
+      }
+
+      // Start timer that checks remaining time.
+      if (remainingSeconds > this.timeWarning) {
+        this.startTimerInterval();
+      }
     });
   }
 
+  copyLia(src: ILia): ILia {
```

Comment:
I had something different in mind. Check this [plunker](http://plnkr.co/edit/hXmgC7d6xtAKKo6m6DtJ?p=preview), specifically `app.ts` and `car.ts` files